app.js

```diff
@@ -155,7 +182,14 @@ io.on('connection', function(socket){
     var remotemenuversion = fs.readFileSync('/config/menuversion.txt', 'utf8');
     var endpointsfile = fs.readFileSync('/config/endpoints.yml');
     var endpoints = yaml.load(endpointsfile);
-    var localfiles = await readdirp.promise('/assets/.');
+    // Wrap readdirp in a promise
+    var localfiles = await new Promise((resolve, reject) => {
+      const entries = [];
+      readdirp('/assets/.')
+        .on('data', (entry) => entries.push(entry))
+        .on('end', () => resolve(entries))
+        .on('error', (error) => reject(error));
+    });
```

Comment:
Addressing the breaking changes that were introduced with version upgrade of `readdirp` to `v4.0.1` from old `v3.6.0`.